src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -322,6 +327,11 @@ instantiatePolyTypeWithUnknowns
   -> m (Expr, SourceType)
 instantiatePolyTypeWithUnknowns val (ForAll _ ident mbK ty _) = do
   u <- maybe (internalCompilerError "Unelaborated forall") freshTypeWithKind mbK
+  case u of
+    TUnknown _ i ->
+      insertUnkName i ident
+    _ ->
+      internalError "freshTypeWithKind didn't produce TUnknown"
```

Comment:
What's the difference between `internalCompilerError` and `internalError`? Should I prefer one over the other in this context?

Reply:
Ooh, I [know this one](https://github.com/purescript/purescript/pull/4126#issuecomment-887694913)!

Reply:
Thanks! I think `internalCompilerError` would be better here, even though `freshTypeWithKind` always returning an unknown is irrefutable.